free-1.md

```diff
@@ -7,44 +7,168 @@ good way to meet their needs.
+been intrigued by its fusion of technology and the arts. I have sat through
+a number of light programming sessions thinking to myself that it is redundent
+and very time consuming. Why does a designer verbally translate what they have
+written on a cue sheet into computer commands? If there was a better langugae
+to do both the description and control, it would be more efficient and easier
+to reflect on without constantly referring to charts and spreadsheets. Furthermore,
+I have stage managed productions where the lighting designed does not make it
+clear to me what each cue is set to do making it more difficult to time correctly.
+
+First semester I was talking with someone about how it would be interesting
+to create more student-led fully lit performances but dreaded the task of making
+worksheets and then spending time with someone especially focused on programming
+lights. I wasn't really sure of another solution, but after thinking about it in
+the context of DSLs I think it would be a really engaging and useful tool.
 
 ### Domain
 _Describe the project's domain in five words._
 
+Easy, expressive theatrical light cueing.
 
 ### Interface (syntax)
-_How might the user interact with the language? What does programming look 
-like? Why is this the right way to interact with the problem domain?_ 
-
+_How might the user interact with the language? What does programming look
+like? Why is this the right way to interact with the problem domain?_
+
+Instead of using an Excel spread sheet and then a lighting console with a
+number pad and keywords, a lighting designer would use a full QWERTY keyboard
+to formally describe the lighting scenes:
+
+```
+production: "DSLs"
+designer: "Ross Wollman <ross.wollman@pomona.edu>"
+zones:
+  center_stage: channel(1)
+  upstage:
+    left: channel(12) channel(81) channel(89)
+    center: channel(18) channel(2) channel(90)
+    right: channel(32) channel(52) channel(18)
+  downstage:
+    left: channel(127)
+    center: channel(123) channel(19)
+    right: channel(34)
+groups:
+  all: zone(upstage) + zone(downstage)
+
+cues:
+  1.0.0: in 1::second downstage::zone goto 100%::intensity
+    to: light the entir stage
+  2.0.0: then goto blackout::intensity
+    to: allow actors to leave
+  3.0.0: after in 2::second downstage:left::goto 20%::intensity
+    to: illuminate the two new actors
+```
+
+The above pseudo code is one way this langugae might look&mdash;the exact
+organization will need to be considered futher, but the overall idea is one
+can create groups and zones (perhaps even paths of lights) that are then
+meaningfully refered to in the `cues` section.
+
+I think this is expressive in that it reads somewhat naturally (although going
+along Fowler's principles, it won't be a natural langugae) and is farily easy
+to communicate. Lighting designers and programmers would likely feel comfortable
+learning this limited langugae.
```

Comment:
Yes&mdash;I imagine some other scheme where you could give a specific channel a variable name. So if channel 12 is aimed at the down stage left corner, you could do something like:
```
dsl::12::channel
```